xarray/core/duck_array_ops.py

```diff
@@ -256,13 +257,20 @@ def as_shared_dtype(scalars_or_arrays, xp=None):
         extension_array_types = [
             x.dtype for x in scalars_or_arrays if is_extension_array_dtype(x)
         ]
-        if len(extension_array_types) == len(scalars_or_arrays) and all(
+        non_nans = [x for x in scalars_or_arrays if not (x is pd.NA or x is np.nan)]
```

Comment:
```suggestion
        non_nans = [x for x in scalars_or_arrays if not isnull(x)]
```

And down. Our `isnull` falls back to `pd.isnull` so it might work.

Reply:
The idea here is to catch the literal value, not just the array.  I suppose arrays that contains all nulls/nans should be considered but I'm not sure how to get around this check for the actual value itself.  

```python
>   non_nans = [x for x in scalars_or_arrays if not isnull(x)]
E   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
Ah, ok we can call `all`, and that handles the two cases.

Reply:
I'd also support a `utils.isna` that check the identity as done now. It's just nice to keep the `pd.*` contained to a handful of places

Reply:
Ah I see your reply, same idea!